### ISSUE TITLE
Accessibility Issues: <ibm-combo-box>

### ISSUE THREAD

**Issue by ajnagashima** [IBM/carbon-components-angular#1560]
## Detailed description

> Describe in detail the issue you're having.

The \<ibm-combo-box\> component fails to pass the accessibility specifications outlined by IBM. There are two violations found for the v3 \<ibm-combo-box\> and issues appear to persist in the v4 Combobox as well. The label created doesn't point to either of the elements given roles; these being the actual \<ibm-combo-box\> given the role "combobox" and an internal div given the role "listbox". 

In some sandbox testing, there appears to be a potential for the `id` property to be duplicated as well, but this could not be verified in an active codebase. 
> Is this a feature request (new component, new icon), a bug, or a general issue?

- Bug
> Is this issue related to a specific component?

- \<ibm-combo-box\>
> What did you expect to happen? What happened instead? What would you like to see changed?

Expected for \<ibm-combo-box\> to pass accessibility scans. The Combobox component failed the following rules:
- Form control with "combobox" role has no associated label
- Form control with "listbox" role has no associated label
> What browser are you working in?

- Chrome
> What version of the Carbon Design System are you using?

- v3. However, some issues appear to persist in v4.
> What offering/product do you work on? Any pressing ship or release dates we should be aware of?

- ibm.com marketplace checkout

## Steps to reproduce the issue

1. [Install IBM accessibility plugin](https://pages.github.ibm.com/checkout/digital-commerce-tribe/development-process/accessibility/).
2. Visit an ibm product page - [example](https://www.ibm.com/marketplace/purchase/configuration/en/us/checkout?editionID=EID5HFZ6).
  a. click the "Continue to checkout" button.
  b. sign in to w3.
3. Start an accessibility scan as described on the accessibility plugin page.
4. Expanding the violations dropdown.
  b. expand the "Each form control must have an associated label" dropdown.
5. Clicking on one of the violations now presented should highlight that portion in the code, showing the \<ibm-combo-box\>

## Possible fixes
Adding `aria` properties to some of the elements removed some of the problems. 

For the main Combobox, adding the property `aria-label` and a label string can fix the issue. Also, adding the `aria-labelledby` property to the Combobox and a corresponding `id` to the label can also fix the issue.

For the nested div, these can also fix the issue where the div doesn't have a corresponding label.

**Comment by Donisius** [comment]
Closing since this doesn't seem to be an issue in v4 and #1761 takes care of it for v3
